### ISSUE TITLE
`No such file or directory: '/usr/bin/qubes-input-trigger'` in dom0 trying to configure `sys-gui-gpu`

### ISSUE THREAD

**Issue by ydirson** [QubesOS/qubes-issues#6893]
Thinking I was just hitting #6789 I checked in `/usr/lib/python3.8/site-packages/qubes/ext/gui.py` that the fix from https://github.com/QubesOS/qubes-core-admin/pull/415 is already present.  However, when I run `qubesctl --all state.highstate` again, even after deleting an unfinished `sys-gui-gpu` first, I still see this issue (from journalctl in dom0):

```
Sep 10 23:21:40 dom0 qubesd[2022]: unhandled exception while calling src=b'dom0' meth=b'admin.vm.Start' dest=b'sys-gui-gpu' arg=b'' len(untrusted_payload)=0
Sep 10 23:21:40 dom0 qubesd[2022]: Traceback (most recent call last):
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib/python3.8/site-packages/qubes/api/__init__.py", line 287, in respond
Sep 10 23:21:40 dom0 qubesd[2022]:     response = yield from self.mgmt.execute(
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib/python3.8/site-packages/qubes/api/admin.py", line 891, in vm_start
Sep 10 23:21:40 dom0 qubesd[2022]:     yield from self.dest.start()
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib/python3.8/site-packages/qubes/vm/qubesvm.py", line 1232, in start
Sep 10 23:21:40 dom0 qubesd[2022]:     await self.fire_event_async('domain-start',
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib/python3.8/site-packages/qubes/events.py", line 236, in fire_event_async
Sep 10 23:21:40 dom0 qubesd[2022]:     effect = task.result()
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib/python3.8/site-packages/qubes/ext/gui.py", line 122, in on_domain_start
Sep 10 23:21:40 dom0 qubesd[2022]:     yield from asyncio.create_subprocess_exec(
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/asyncio/subprocess.py", line 236, in create_subprocess_exec
Sep 10 23:21:40 dom0 qubesd[2022]:     transport, protocol = await loop.subprocess_exec(
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/asyncio/base_events.py", line 1630, in subprocess_exec
Sep 10 23:21:40 dom0 qubesd[2022]:     transport = await self._make_subprocess_transport(
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/asyncio/unix_events.py", line 197, in _make_subprocess_transport
Sep 10 23:21:40 dom0 qubesd[2022]:     transp = _UnixSubprocessTransport(self, protocol, args, shell,
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/asyncio/base_subprocess.py", line 36, in __init__
Sep 10 23:21:40 dom0 qubesd[2022]:     self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/asyncio/unix_events.py", line 789, in _start
Sep 10 23:21:40 dom0 qubesd[2022]:     self._proc = subprocess.Popen(
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/subprocess.py", line 858, in __init__
Sep 10 23:21:40 dom0 qubesd[2022]:     self._execute_child(args, executable, preexec_fn, close_fds,
Sep 10 23:21:40 dom0 qubesd[2022]:   File "/usr/lib64/python3.8/subprocess.py", line 1704, in _execute_child
Sep 10 23:21:40 dom0 qubesd[2022]:     raise child_exception_type(errno_num, err_msg, err_filename)
Sep 10 23:21:40 dom0 qubesd[2022]: FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/qubes-input-trigger'
```

It looks like it is trying to execute this program in dom0, where there is no such script, which can easily be checked with a `touch /usr/bin/qubes-input-trigger`, which as can be expected turns this error into `Permission denied: '/usr/bin/qubes-input-trigger'`.

Is that script missing from dom0, or is there something that tries to run it there by mistake ?  `rpm -qf` does not seem to believe it would be missing from an installed package.

**Comment by andrewdavidwong** [comment]
> Thinking I was just hitting #6789 I checked in `/usr/lib/python3.8/site-packages/qubes/ext/gui.py` that the fix from [QubesOS/qubes-core-admin#415](https://github.com/QubesOS/qubes-core-admin/pull/415) is already present. However, when I run `qubesctl --all state.highstate` again, even after deleting an unfinished `sys-gui-gpu` first, I still see this issue (from journalctl in dom0): [...]

Are you sure this isn't a duplicate of #6789? It sounds like the purported fix from QubesOS/qubes-core-admin#415 just doesn't fully fix the bug reported in #6789, which sounds like the same bug being reported here. Let me know if I'm mistaken.

------

This appears to be a [duplicate](https://www.qubes-os.org/doc/issue-tracking/#new-issues-should-not-be-duplicates-of-existing-issues) of an existing issue. If so, please comment on the appropriate existing issue instead. If you believe this is not really a duplicate, please leave a comment briefly explaining why. We'll be happy to take another look and, if appropriate, reopen this issue. Thank you.

**Comment by andrewdavidwong** [comment]
Duplicate of #6789

**Comment by ydirson** [comment]
> Are you sure this isn't a duplicate of #6789?

Yes, I'm sure it is not a duplicate. I confirm the command string did need to be split, and that the merged patch does fix it.

But the problem I have here is clearly different.

**Comment by andrewdavidwong** [comment]
> > Are you sure this isn't a duplicate of #6789?
> 
> Yes, I'm sure it is not a duplicate. I confirm the command string did need to be split, and that the merged patch does fix it.
> 
> But the problem I have here is clearly different.

Ok, reopened.

**Comment by andrewdavidwong** [comment]
@marmarek asked in https://github.com/QubesOS/qubes-issues/issues/6789#issuecomment-917256875:
> Is `qubes-input-proxy-sender` package installed in dom0?

**Comment by ydirson** [comment]
> @marmarek asked in [#6789 (comment)](https://github.com/QubesOS/qubes-issues/issues/6789#issuecomment-917256875):
> 
> > Is `qubes-input-proxy-sender` package installed in dom0?

Probably not :)

However I'll be able to answer only after recovering -- see https://forum.qubes-os.org/t/disabling-sys-gui-gpu-from-commandline/6288, I have hacked around the missing script to see how far it would do (symlinking to /bin/true or /bin/false allowed to proceed with the installation, without one making a difference on the other). Currently trying to find a boot media that will allow me to run qubes-prefs...

**Comment by ydirson** [comment]
> Is `qubes-input-proxy-sender` package installed in dom0?

Definitely not, dom0 only has `qubes-input-proxy` and `qubes-input-proxy-receiver`.  How is `-sender` supposed to get installed ? (this machine had a 4.0.4 install, later upgraded to 4.1beta).

If I install it, the sys-gui-gpu install does seem to work (no more no less than when I had a `/bin/false` symlink)

**Comment by marmarek** [comment]
It seems to be missing in https://github.com/QubesOS/qubes-mgmt-salt-dom0-virtual-machines/blob/master/qvm/sys-gui-gpu.sls (should be there).
